frontend/src/metabase/actions/components/ActionForm/utils.ts

```diff
@@ -143,3 +186,27 @@ export const getFormValidationSchema = (
     });
   return Yup.object(Object.fromEntries(schema));
 };
+
+export const reorderFields = (
+  fields: FieldSettingsMap,
+  oldIndex: number,
+  newIndex: number,
+) => {
+  // we have to jump through some hoops here because fields settings are an unordered map
+  // with order properties
+  const fieldsWithIds = _.mapObject(fields, (field, key) => ({
```

Comment:
nit: idk about browser support and preferences about underscore usage, but this `mapObject` seems can be replaced with 

```
Object.fromEntries(Object.entries(fields).map(([key, value]) => ({...value, id: key}))
```

Reply:
Overall, I think we're trying to avoid underscore if there's a built-in JavaScript way for doing the same thing (e.g., `list.filter(fn)` instead of `_.filter(list, fn)`). Doesn't seem like we can easily apply this to `mapObject`, though. Anyway, I don't have a strong preference, would love to hear what others think as well